File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    deliveries = deliveries[::-1] # 배달 배열 역순 정렬
    pickups = pickups[::-1] # 픽업 배열 역순 정렬
    
    giveBox = 0
    receiveBox = 0
    
    for i in range(n):
        giveBox += deliveries[i]
        receiveBox += pickups[i]
        while giveBox > 0 or receiveBox > 0:
            giveBox -= cap
            receiveBox -= cap
            answer += (n - i) * 2
    return answer

File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.js
function solution(cap, n, deliveries, pickups) {
    var answer = 0;
    // 배달 배열, 수거 배열 역순 정렬
    deliveries.reverse();
    pickups.reverse();
    
    let deli = 0;
    let pick = 0;
    
    // 배열을 돌면서 deli 와 pick이 양수일때 answer에 곱해줌
    for (let i = 0; i < n; i ++) {
        deli += deliveries[i];
        pick += pickups[i];
        
        while (deli > 0 || pick > 0) {
            deli -= cap;
            pick -= cap;
            answer += (n - i) * 2
        };
    };
    
    return answer;
}